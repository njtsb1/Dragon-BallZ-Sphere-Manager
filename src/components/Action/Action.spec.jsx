import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Action from './Action'

//mocks
import { profile } from '../../mocks/profile.json';
import profileSuccess from '../../mocks/profileSuccess.json';

it('Should render Action', () => {
  const { container, debug, getByText } = render(<Action balls={profile.balls} />)
  const button = getByText('Summon up').closest('button')

  expect(getByText('Summon up shenron')).toBeInTheDocument()
  expect(button).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})

it('Should open modal if user dont have all dragon balls', () => {
  const { debug, getByText } = render(<Action balls={profile.balls} />)
  const button = getByText('Summon up').closest('button')

  fireEvent.click(button)
  expect(getByText('You dont have all spheres to summon shenron')).toBeInTheDocument()

  const backButton = getByText('Back').closest('button')
  fireEvent.click(backButton)

  expect(getByText('Summon up shenron')).toBeInTheDocument()
})

it('Should show shenron', () => {
  const { container, debug, getByText, getByTestId } = render(<Action balls={profileSuccess.profile.balls} />)
  const button = getByText('Summon up').closest('button')

  fireEvent.click(button)
  expect(getByTestId('shenron')).toBeInTheDocument()

  expect(container).toMatchSnapshot()
})


