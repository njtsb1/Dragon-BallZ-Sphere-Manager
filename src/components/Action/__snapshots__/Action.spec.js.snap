// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Action 1`] = `
<div>
  <div
    class="sc-AxjAm gqoJub"
    data-testid="card-shenron"
  >
    <div
      class="css-1brk784"
    >
      Summon up shenron
    </div>
    <button
      class="btn btn-warning"
      data-testid="invoke-button"
      type="button"
    >
      Summon up
    </button>
  </div>
</div>
`;

exports[`Should show shenron 1`] = `
<div>
  <img
    class="sc-AxirZ hyvxts"
    data-testid="shenron"
    src="https://static.wixstatic.com/media/ecfb5e_dbd3be9fe8f14cc2a37e91167d125bc2.png/v1/fill/w_784,h_784,al_c,q_95,usm_0.66_1.00_0.01/ecfb5e_dbd3be9fe8f14cc2a37e91167d125bc2.webp"
  />
  <div
    class="sc-AxjAm gqoJub"
    data-testid="card-shenron"
  >
    <div
      class="css-1brk784"
    >
      Summon up shenron
    </div>
    <button
      class="btn btn-warning"
      data-testid="invoke-button"
      type="button"
    >
      Summon up
    </button>
  </div>
</div>
`;
